components/editor/XmlViewer.tsx

```
@@ -0,0 +1,319 @@
+                /(&gt;)([^&<]+)(&lt;)/g,
+                '$1<span class="text-gray-800 dark:text-gray-200">$2</span>$3'
+            );
+
+            return (
+                <div key={index} className="flex hover:bg-muted/30 transition-colors">
+                    <span className="text-gray-400 text-xs mr-4 select-none w-8 text-right flex-shrink-0 py-0.5">
+                        {lineNumber}
+                    </span>
+                    <span
+                        className="flex-1 min-w-0 py-0.5 leading-relaxed font-mono"
+                        dangerouslySetInnerHTML={{ __html: highlightedLine }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)